src/nnn.c

```diff
@@ -7969,7 +7970,7 @@ static char *load_input(int fd, const char *path)
 	} else
 		xstrsncpy(cwd, path, PATH_MAX);
 
-	while ((chunk_count) < 512 && !msgnum) {
+	while (chunk_count < LIST_INPUT_MAX / chunk && !msgnum) {
 		input_read = read(fd, input + total_read, chunk);
 		if (input_read < 0) {
 			DPRINTF_S(strerror(errno));
```

Comment:
I suppose we can take this as an opportunity to make this more robust against signals.

```suggestion
		if (input_read < 0) {
			if (errno == EINTR)
				continue;
			DPRINTF_S(strerror(errno));
```